cellxgene_schema_cli/cellxgene_schema/atac_seq.py

```diff
@@ -257,10 +257,24 @@ def report_errors(header: str, errors: list[str]) -> list[str]:
 
 
 def validate_anndata(anndata_file: str) -> list[str]:
-    errors = [validate_anndata_organism_ontology_term_id(anndata_file), validate_anndata_is_primary_data(anndata_file)]
+    errors = [
+        validate_anndata_organism_ontology_term_id(anndata_file),
+        validate_anndata_is_primary_data(anndata_file),
+        validate_anndata_raw_counts(anndata_file),
+    ]
     return report_errors("Errors found in Anndata file. Skipping fragment validation.", errors)
 
 
+def validate_anndata_raw_counts(anndata_file: str) -> Optional[str]:
+    # check if the raw counts are required and present
+    with h5py.File(anndata_file) as f:
+        assay_ontology_term_ids = ad.io.read_elem(f["obs"])["assay_ontology_term_id"]
+        if is_atac(assay_ontology_term_ids.iloc[0]) != "p":
+            return
+        if "raw/X" not in f:
```

Comment:
We can't assume no raw/X == no raw counts. We must validate if X is a raw count matrix, and fail if both no raw.X AND X is not raw counts. 

That said, I think we already validate for this in the anndata validation. Not sure if there's an extra nuance to checking here or not, my understanding is that in practice this will always run in tandem with anndata validation so not sure if we need to add it here (someone else should validate whether that is a fair assumption on my part)